src/lib.rs

```diff
@@ -56,9 +57,20 @@ impl Cli {
 
         let delete_file = run::generate_delete_file(tmp_dir.path())?;
 
+        let subprocesses = config
+            .subprocesses
+            .iter()
+            .filter_map(|x| x.spawn(tmp_dir.path()))
+            .collect::<Vec<process::Child>>();
+
         run::start_shell(config, tmp_dir.path())?;
```

Comment:
This is wrong. Imagine the function `start_shell()` would fail, what should happen to the subprocesses? Here you're not doing anything with them so they will not be cleaned up